spec/spec_helper.rb

```diff
@@ -24,6 +26,8 @@
 
 Dir[File.join(Dir.pwd, 'lib', 'token_validator.rb')].sort.each { |file| require file }
 
+WebMock.disable_net_connect!(allow: 'https://api.codacy.com')
```

Comment:
Why is this needed?

Reply:
So that we can send the coverage report to Codacy without the network request being blocked by WebMock. The codecov gem disables WebMock and VCR before sending coverage reports, but the Codacy coverage gem does not do this, so we have to manually disable WebMock and/or VCR manually.